### ISSUE TITLE
Clean up the edit layout interface

### ISSUE THREAD

**Issue by jenlampton** [backdrop/backdrop-issues#373]
Based on feedback from https://github.com/backdrop/backdrop-issues/issues/86

We need to clean up the edit layout interface, but none of the following issues were important enough to hold up the initial patch.

1)  It's not clear how much of the page is the layout, and how much is just the rest of the admin page.  There's a faint gray background on some of the page, but that's not enough.  I wonder if adding a solid border around that area of the page might help.  (I also think getting rid of stuff that's not part of the layout might also help - the title type selection needs to go!).  Here's an attempt at adding a border around the "layout" area. (feedback please!)
![layout-border](https://cloud.githubusercontent.com/assets/397895/4674241/b97530bc-55b2-11e4-9ffd-1624c005e0f0.png)

2) The title type selection needs to go. All the other configurations are in modals, so I would love to see the "title" area of the page have a similar configure link, and a modal, where we can hide this title type business. (see screenshot below)

3) The page title needs to look more like a page title. As it is, it totally gets lost on the page. For starters, it needs to be a H1 instead of a H2. It should probably also be an "area" and not just a string of text, this can be accomplished with a border (and/or a background), like so:
![layout-title](https://cloud.githubusercontent.com/assets/397895/4674480/6a0418ea-55b7-11e4-9b2e-3b0f2043f3b6.png)

4) The "add block" buttons are ambiguous as to which region they are adding the block into. Yes, there is a label next to each button, but we don't need to label things, we can just show them.  Let's move the labels and buttons inside the region that will hold the block. And add some more spacing. Like so.
![layout-show-region](https://cloud.githubusercontent.com/assets/397895/4674583/1d53b092-557f-11e4-9f6f-65af59e06535.png)

**Comment by klonos** [comment]
What's left here after #891 ??

**Comment by quicksketch** [comment]
I think items 2 and 3 still need to be addressed, both relating to the title configuration. Let's split this out into a dedicated issue: https://github.com/backdrop/backdrop-issues/issues/952
